dpnp/backend/kernels/dpnp_krnl_sorting.cpp

```diff
@@ -83,6 +83,55 @@ struct _sort_less
     }
 };
 
+template <typename _DataType>
+class dpnp_partition_c_kernel;
+
+template <typename _DataType>
+void dpnp_partition_c(const void* sort_array1_in, void* result1, const size_t kth, const size_t* shape, const size_t ndim)
+{
+
+    cl::sycl::event event;
+
+    const _DataType* sort_arr = reinterpret_cast<const _DataType*>(sort_array1_in);
+    _DataType* result = reinterpret_cast<_DataType*>(result1);
+
+    size_t size_ = 1;
+    for (size_t i = 0; i < ndim - 1; ++i)
```

Comment:
`ndim - 1` is suspicious for me. It looks like `size_` represents number of elements in the array. If `ndim==1` the variable `size` will have `1` instead `shape[0]`